packages/@tinacms/graphql/src/resolve.ts

```diff
@@ -409,6 +304,20 @@ export const resolve = async ({
             // `unionData` is used by the typeResolver, need to keep this check in-place
             // This is an array in many cases so it's easier to just pass it through
             // to be handled by the `typeResolver`
+            if (!value) {
```

Comment:
Previously `unionData` was only called when we already had values. Because, like I mentioned in another comment, the "union" was on the `data` key, and by the time GraphQL fetches the "data" property we've already resolved the document. But now, we run into this union 1 step earlier, so the value is `null`. It's not null when it comes from a "connection" query, though, so the conditional check is still needed.